Async.Model/AsyncLoaded/ThreadSafeAsyncLoader.cs

```diff
@@ -36,12 +37,37 @@ public override void Conj(TItem item)
             }
         }
 
+        public override void Replace(TItem oldItem, TItem newItem)
+        {
+            using (mutex.Lock())
+            {
+                seq.Replace(oldItem, newItem);
+                NotifyCollectionChanged(ChangeType.Updated, newItem);
+            }
+        }
+
         public override void ReplaceAll(IEnumerable<TItem> newItems)
         {
             using (mutex.Lock())
             {
-                base.ReplaceAll(newItems);
+                seq.ReplaceAll(newItems);
+
+                // TODO: This will generate updates for unchanged items, is this acceptable?
```

Comment:
At the moment, development speed is of the essence, so for now I think it is ...?